src/react_native/utils.cljs

```diff
@@ -37,3 +40,24 @@
         props       (when (map? first-child) first-child)
         children    (if props (rest argv) argv)]
     [props children]))
+
+(defn convert-prop-value
+  "Based on `reagent.impl.template/kv-conv`.
+  Takes the prop map of a reagent component and returns a React-valid property JS Object
+  by transforming kebab-case keys -> camelCase, maps -> JS Objects and vectors -> arrays.
+
+  This version adds support to recursively transform properties inside vectors, to have a
+  more consistent developer experience in React Native."
+  [x]
+  (cond
+    (reagent.util/js-val? x) x
+    (reagent.util/named? x)  (name x)
+    (map? x)                 (reduce-kv template/kv-conv #js {} x)
+    (vector? x)              (transforms/map-array convert-prop-value x)
+    (coll? x)                (clj->js x)
+    (ifn? x)                 (fn [& args]
+                               (apply x args))
+    :else                    (clj->js x)))
+
+;; We override the default reagent implementation with the one that supports vectors.
+(set! template/convert-prop-value convert-prop-value)
```

Comment:
Will `template/kv-conv` call `template/convert-prop-value`?

Reply:
Hey @seanstrom !

`template/kv-conv` will call our version of `convert-prop-value`, that's why we use `set!`.
You can take a look at the tests where I show that the transformation is performed recursively with nested vectors.

> Just want to make sure if maps are still handled the same.

What do you mean by "the same"? now they support vectors inside, besides that, they should behave in the same way.

Reply:
Oh good! I was just wanted to understand if we're using `convert-prop-value` for maps, which it seems we are based on you mentioned about setting `template/convert-prop-value`. Very cool 🙌

Reply:
Yes, we are using it for maps too. Actually for every prop conversion. This version is pretty similar to what reagent originally does